ydb/core/persqueue/pq_impl.cpp

```diff
@@ -3915,6 +4223,16 @@ void TPersQueue::ProcessSourceIdRequests(ui32 partitionId) {
     }
 }
 
+void TPersQueue::Handle(NLongTxService::TEvLongTxService::TEvLockStatus::TPtr& ev, const TActorContext& /*ctx*/)
+{
+    auto& record = ev->Get()->Record;
+    ui64 writeId = record.GetLockId();
+    if (TxWrites.contains(writeId)) {
```

Comment:
If  lockStatus == NO TX and Writeid is presented in memory state  and no propose in memory state - delete it and all partitions

Reply:
Change deletion of writeID to :
1) Move write id to  inmemory map to_be_deleted_write_id -> shadow_partitions
2) Send to all shadow partitions event ClearAllAndDie, wait for results
3) when all results for partitions in writeId received, shedule deletion of all info for write id in tablet state.

Reply:
We agreed to do it in a separate task. There I described the above algorithm
LOGBROKER-8871